src/http/gateway/resources/gateway.js

```diff
@@ -120,6 +121,15 @@ module.exports = {
 
         let response = reply(stream2).hold()
 
+        // Etag maps directly to an identifier for a specific version of a resource
+        // TODO: change to .cid.toBaseEncodedString() after switch to new js-ipfs-http-response
+        response.header('Etag', `"${data.multihash}"`)
```

Comment:
is `data.multihash` already a string?

Reply:
Yes, it is a string already. `Etag` spec [requires wrapping it with `""`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag#Directives)

Note: the source of this value may change after https://github.com/ipfs/js-ipfs-http-response/pull/9 lands  – it will add CID object as `data.cid`. It just needs to be a unique string, so we can use cid or a raw multihash – both are fine for this.